### PR TITLE
Fix warnings when building docs

### PR DESCRIPTION
**Pull request by davidwagner**
**Changes proposed:**
Updates the tests to eliminate most of the warnings when running `make docs`, and clean up some outdated/obsolete links.  Partly addresses #357.

### CODE REVIEW

File: datascience/tables.py [hunk 1]
```diff
@@ -1456,12 +1459,10 @@ def sample(self, k=None, with_replacement=True, weights=None):
         job  | wage
         a    | 10
         a    | 10
-        >>> jobs.sample(k=2, weights=make_array(1, 0, 1, 0))
+        >>> jobs.sample(k=2, weights=make_array(1, 0, 1, 0)) # Weights must be length of table.
```
Comment (adnanhemani) [inline datascience/tables.py]: I think we shouldn't put the comment at the end of this line. Usually comments in doctests where the doctest is failing is meant to be a pointer for a common pitfall - but the error here is different ("probabilities do not sum to 1") than what's in the comment

File: docs/tutorial.rst [hunk 2]
```diff
@@ -363,16 +362,16 @@ Let's do the bootstrap test on the two categories.
...
         """
         Computes one bootstrapped difference in means.
         The table.sample method lets us take random samples.
         We then split according to the number of nonsmokers in the original sample.
         """
         resample = smoker_and_wt.sample(with_replacement = True)
-        bootstrap_diff = resample.column('birthwt')[:num_nonsmokers].mean() - \
-            resample.column('birthwt')[num_nonsmokers:].mean()
+        bootstrap_diff = resample.column('Birth Weight')[:num_nonsmokers].mean() - \
+            resample.column('Birth Weight')[num_nonsmokers:].mean()
```
Comment (adnanhemani) [inline docs/tutorial.rst]: I'm not familiar with how to deploy this locally (due to it being a Rust file), but the changes made seem to be 100% correct to make. Let's deploy and verify!

File: datascience/tables.py (hunk 1 above)
Comment (davidwagner) [reply datascience/tables.py]: Oh gosh yes, that was dumb!  Fixed.  Thanks.

File: docs/tutorial.rst (hunk 2 above)
Comment (davidwagner) [reply docs/tutorial.rst]: I built the docs with `make docs`, `make serve_docs` and then viewed in my browser at `localhost:8000` and the result looks ok, so I'm guessing this is probably ok.

**Comment by davidwagner** [comment]
This is ready for you to re-review.  Thank you!

**Pull request merged**